From 05e6737c4db7619a8a17eb29c7f53755677f3ea9 Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Mon, 5 May 2014 15:02:04 -0500
Subject: [PATCH 11/21] remoteproc/pruss: define platform data for PRU
 remoteproc

The PRUSS has a PRCM reset line associated with the IP. The reset
needs to be programmed properly before accessing any of the internal
registers in the PRUSS. This functionality is achieved through the
omap_device layer, which is not exposed outside of mach-omap2 layer.
Define a platform data structure for PRUSS so that this API can be
invoked through platform data ops from the driver.

Signed-off-by: Suman Anna <s-anna@ti.com>
---
 include/linux/platform_data/remoteproc-pruss.h |   33 ++++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 include/linux/platform_data/remoteproc-pruss.h

diff --git a/include/linux/platform_data/remoteproc-pruss.h b/include/linux/platform_data/remoteproc-pruss.h
new file mode 100644
index 0000000..877d548
--- /dev/null
+++ b/include/linux/platform_data/remoteproc-pruss.h
@@ -0,0 +1,33 @@
+/*
+ * Platform data for PRUSS on TI SoCs
+ *
+ * Copyright (C) 2014 Texas Instruments, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef _PLAT_REMOTEPROC_PRUSS_H
+#define _PLAT_REMOTEPROC_PRUSS_H
+
+struct platform_device;
+
+/**
+ * struct pruss_platform_data - PRUSS remoteproc's platform data
+ * @reset_name: name of the reset
+ * @assert_reset: PRU-specific handler for putting the device in reset
+ * @deassert_reset: PRU-specific handler for releasing the device from reset
+ */
+struct pruss_platform_data {
+	const char *reset_name;
+	int (*assert_reset)(struct platform_device *pdev, const char *name);
+	int (*deassert_reset)(struct platform_device *pdev, const char *name);
+};
+
+#endif /* _PLAT_REMOTEPROC_PRUSS_H */
-- 
1.7.9.5

